runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -1841,6 +1841,52 @@ handleServerMessage(JITServer::ClientStream *client, TR_J9VM *fe, JITServer::Mes
          client->write(response, objLocation, obj);
          }
          break;
+      case MessageType::ResolvedMethod_getResolvedImplementorMethods:
+         {
+         auto recv = client->getRecvData<TR_ResolvedJ9Method *, std::vector<TR_OpaqueClassBlock *>, int32_t, bool>();
+         auto owningMethod = std::get<0>(recv);
+         auto &subClasses = std::get<1>(recv);
+         int32_t cpIndexOrOffset = std::get<2>(recv);
+         bool isInterface = std::get<3>(recv);
+
+         int32_t numMethods = subClasses.size();
+         std::vector<J9Method *> ramMethods;
+         std::vector<TR_ResolvedJ9JITServerMethodInfo> methodInfos;
+         ramMethods.reserve(numMethods);
+         methodInfos.reserve(numMethods);
+
+         for (int32_t i = 0; i < numMethods; ++i)
+            {
+            J9Method *ramMethod;
+            TR_ResolvedJ9JITServerMethodInfo methodInfo;
+            if (isInterface)
+               {
+               ramMethod = (J9Method *) fe->getResolvedInterfaceMethod(owningMethod->getPersistentIdentifier(), subClasses[i], cpIndexOrOffset);
+               bool resolved = ramMethod && J9_BYTECODE_START_FROM_RAM_METHOD(ramMethod);
+
+               if (resolved)
+                  TR_ResolvedJ9JITServerMethod::createResolvedMethodFromJ9MethodMirror(methodInfo, (TR_OpaqueMethodBlock *) ramMethod, 0, owningMethod, fe, trMemory);
+               }
+            else
+               {
+               TR_OpaqueMethodBlock *method = fe->getResolvedVirtualMethod(subClasses[i], cpIndexOrOffset);
+               ramMethod = reinterpret_cast<J9Method *>(method);
+               TR_ResolvedJ9JITServerMethodInfo methodInfo;
+               if (ramMethod)
+                  TR_ResolvedJ9JITServerMethod::createResolvedMethodMirror(methodInfo, method, 0, owningMethod, fe, trMemory);
+               }
+            ramMethods.push_back(ramMethod);
```

Comment:
Yes it's intended. This will cache the first unresolved method on the server, so that when the actual tree walk is done, it will know to stop the walk there.